src/java/com/unifina/data/RealtimeEventQueue.java

```diff
@@ -3,123 +3,72 @@
 import com.unifina.datasource.DataSource;
 import com.unifina.datasource.DataSourceEventQueue;
 import com.unifina.exceptions.StreamFieldChangedException;
-import com.unifina.signalpath.StopRequest;
 import com.unifina.utils.Globals;
 import org.apache.log4j.Logger;
 
-import java.util.ArrayDeque;
-import java.util.Queue;
+import java.util.concurrent.TimeUnit;
 
-public class RealtimeEventQueue extends DataSourceEventQueue implements IEventRecipient {
+public class RealtimeEventQueue extends DataSourceEventQueue {
 	private static final Logger log = Logger.getLogger(RealtimeEventQueue.class);
-	private static final int LOGGING_INTERVAL = 10000; // set to 0 for no logging
 
 	private boolean firstEvent = true;
-	private EventQueueMetrics eventQueueMetrics = new RealTimeEventQueueMetrics();
 
 	public RealtimeEventQueue(Globals globals, DataSource dataSource) {
-		super(true, globals, dataSource);
+		this(globals, dataSource, DEFAULT_CAPACITY);
 	}
 
-	@Override
-	protected Queue<FeedEvent> initQueue() {
-		return new ArrayDeque<>();
+	public RealtimeEventQueue(Globals globals, DataSource dataSource, int capacity) {
+		super(globals, dataSource, capacity);
 	}
 
-	protected void doStart() throws Exception {
-		log.info("The realtime event queue is starting!");
-
-		int eventCounter = 0;
-		long elapsedTime = 0;
+	@Override
+	protected void runEventLoopUntilDone() {
+		log.info("runEventLoopUntilDone: starting!");
 
 		while (!isAborted()) {
-			FeedEvent event = waitForAndPullFeedEvent();
+			Event event = null;
+			try {
+				event = queue.poll(1, TimeUnit.SECONDS);
+			} catch (InterruptedException e) {
+				log.error(e);
+			}
+
 			if (event == null) {
-				log.info("doStart: aborting");
-				return;
+				// Timed out while waiting for new events. Just keep trying until aborted.
+				continue;
 			}
```

Comment:
I spent some time on this and cleaned it up. It actually lead to `RealtimeEventQueue` becoming empty/trivial. So I deleted it and instead of three classes I ended up with just two:
- `DataSourceEventQueue`, used by `RealtimeDataSource`
- `HistoricalEventQueue extends DataSourceEventQueue`, used by `HistoricalDataSource`